migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs

```diff
@@ -462,73 +433,28 @@ impl<'schema> SqlSchemaDiffer<'schema> {
     }
 }
 
-fn push_previous_usages_as_defaults_in_altered_enums(differ: &SqlSchemaDiffer<'_>, alter_enums: &mut [AlterEnum]) {
-    for alter_enum in alter_enums {
-        let mut previous_usages_as_default = Vec::new();
-
-        let enum_names = differ.schemas.enums(&alter_enum.index).map(|enm| enm.name());
-
-        for table in differ.dropped_tables() {
-            for column in table
-                .columns()
-                .filter(|col| col.column_type_is_enum(enum_names.previous()) && col.default().is_some())
-            {
-                previous_usages_as_default.push(((column.table().table_id(), column.column_id()), None));
-            }
-        }
-
-        for tables in differ.table_pairs() {
-            for column in tables
-                .dropped_columns()
-                .filter(|col| col.column_type_is_enum(enum_names.previous()) && col.default().is_some())
-            {
-                previous_usages_as_default.push(((column.table().table_id(), column.column_id()), None));
-            }
-
-            for columns in tables.column_pairs().filter(|col| {
-                col.previous.column_type_is_enum(enum_names.previous()) && col.previous.default().is_some()
-            }) {
-                let next_usage_as_default = Some(&columns.next)
-                    .filter(|col| col.column_type_is_enum(enum_names.next()) && col.default().is_some())
-                    .map(|col| (col.table().table_id(), col.column_id()));
-
-                previous_usages_as_default.push((
-                    (columns.previous.table().table_id(), columns.previous.column_id()),
-                    next_usage_as_default,
-                ));
-            }
-        }
-
-        alter_enum.previous_usages_as_default = previous_usages_as_default;
-    }
-}
-
 /// Compare two [ForeignKey](/sql-schema-describer/struct.ForeignKey.html)s and return whether they
 /// should be considered equivalent for schema diffing purposes.
-fn foreign_keys_match(fks: Pair<&ForeignKeyWalker<'_>>, flavour: &dyn SqlFlavour) -> bool {
+fn foreign_keys_match(fks: Pair<&ForeignKeyWalker<'_>>, flavour: &dyn SqlFlavour, db: &DifferDatabase<'_>) -> bool {
```

Comment:
You don't need both: the flavour and the database in here!

Reply:
Good catch!